lib/checks/canAddLikeTermWithCoefficientNodes.js

```diff
@@ -0,0 +1,35 @@
+const Node = require('../node');
+
+// Returns true if the nodes are terms with coefficients that can be added together.
+// The nodes need to have the same base and exponent
+// e.g. 2x + 5x, 6x^2 + x^2, nthRoot(4,2) + nthRoot(4,2)
+function canAddLikeTermWithCoefficientNodes(node, termWithCoefficientSubclass) {
+  if (!Node.Type.isOperator(node) || node.op !== '+') {
+    return false;
+  }
+  const args = node.args;
+  if (!args.every(n => Node.TermWithCoefficient.isTermWithCoefficient(
+    n, termWithCoefficientSubclass.baseNodeFunc))) {
+    return false;
+  }
+  if (args.length === 1) {
```

Comment:
So I tried this and it broke some stuff -- seems like in our nthRoot code, we have some situations where isOperator is true but args.length = 0 or 1. I went down a rabbit hole trying to figure it out, but it was super tricky and seemed out of the scope of this PR. So, leaving the check here for now!